sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveCatalog.scala

```diff
@@ -85,6 +85,13 @@ private[spark] class HiveCatalog(client: HiveClient) extends ExternalCatalog wit
     withClient { getTable(db, table) }
   }
 
+  /**
+   * Switch our client to one that belongs to the new session.
+   */
+  def newSession(newClient: HiveClient): this.type = {
```

Comment:
just talked to @yhuai offline. I don't think we need this. The whole reason why I did something like this is because `HiveClient` also has session-specific stuff, but as long as we use the root `HiveClient` (the first one) then we're fine.